packages/core/src/browser/preferences/preference-contribution.ts

```diff
@@ -96,6 +84,23 @@ export class PreferenceSchemaProvider extends PreferenceProvider {
         return this.preferences;
     }
 
+    setSchema(schema: PreferenceSchema): void {
+        const props: string[] = [];
+        for (const property in schema.properties) {
+            if (this.combinedSchema.properties[property]) {
+                console.error('Preference name collision detected in the schema for property: ' + property);
+            } else {
+                this.combinedSchema.properties[property] = schema.properties[property];
+                props.push(property);
+            }
+        }
+        this.validateFunction = new Ajv().compile(this.combinedSchema);
```

Comment:
please refactor that it is called only once from the constructor, not for each contribution

Reply:
Thanks for the review!
My point here is to be able to validate a plugin schema too. Do we have another way to validate it?

Reply:
sure, you can call it from `setSchema` as well

just refactor it in a way that it is called once from the constructor and on each call of `setSchema`

Reply:
It is already called once from the constructor since `setSchema` is called there. Or maybe I missed something?

Reply:
it is called in `forEach` loop for each contribution

Reply:
fixed